src/titleui/campaign.cpp

```diff
@@ -0,0 +1,2813 @@
+	bool isTruncated = false;
+};
+
+int32_t WZOptionTitleLabel::idealWidth()
+{
+	return wzText.width();
+}
+
+WzString WZOptionTitleLabel::getString() const
+{
+	return wzText.getText();
+}
+
+void WZOptionTitleLabel::setString(WzString string)
+{
+	wzText.setText(string, font_regular);
+}
+
+std::string WZOptionTitleLabel::getTip()
+{
+	if (!isTruncated)
+	{
+		return "";
+	}
+	return wzText.getText().toUtf8();
+}
+
+void WZOptionTitleLabel::display(int xOffset, int yOffset)
+{
+	int x0 = x() + xOffset;
+	int y0 = y() + yOffset;
+
+	PIELIGHT textColor = WZCOL_TEXT_MEDIUM;
+
+	int textX0 = x0;
+	int textY0 = y0 + verticalPadding - wzText.aboveBase();
+
+	int maxTextDisplayableWidth = width();
+	isTruncated = maxTextDisplayableWidth < wzText.width();
+	if (isTruncated)
+	{
+		maxTextDisplayableWidth -= (iV_GetEllipsisWidth(wzText.getFontID()) + 2);
+	}
+	wzText.render(textX0, textY0, textColor, 0.0f, maxTextDisplayableWidth);
+	if (isTruncated)
+	{
+		// Render ellipsis
+		iV_DrawEllipsis(wzText.getFontID(), Vector2f(textX0 + maxTextDisplayableWidth + 2, textY0), textColor);
+	}
+}
+
+static std::shared_ptr<WZOptionTitleLabel> makeCampaignUIOptionTitleLabel(const WzString& text, iV_fonts font)
+{
+	auto psLabel = std::make_shared<WZOptionTitleLabel>();
+	psLabel->setFont(font);
+	psLabel->setString(text);
+	psLabel->setGeometry(0, 0, psLabel->idealWidth(), iV_GetTextLineSize(font));
+	return psLabel;
+}
+
+void CampaignStartOptionsForm::initialize(const std::shared_ptr<WzCampaignSelectorTitleUI>& parent)
+{
+	auto weakParent = std::weak_ptr<WzCampaignSelectorTitleUI>(parent);
+	auto weakSelf = std::weak_ptr<CampaignStartOptionsForm>(std::dynamic_pointer_cast<CampaignStartOptionsForm>(shared_from_this()));
+
+	titleArea = WzCampaignTitleWidget::make(modInfo);
+	attach(titleArea);
+	titleArea->setCalcLayout([](WIDGET *psWidget) {
+		auto psParent = psWidget->parent();
+		ASSERT_OR_RETURN(, psParent != nullptr, "No parent?");
+		psWidget->setGeometry(0, 0, psParent->width(), 155);
+	});
+
+	subInfoBox = WzCampaignSubInfoBox::make(modInfo);
+	attach(subInfoBox);
+	subInfoBox->setCalcLayout([](WIDGET *psWidget) {
+		auto psParent = std::dynamic_pointer_cast<CampaignStartOptionsForm>(psWidget->parent());
+		ASSERT_OR_RETURN(, psParent != nullptr, "No parent?");
+		int y0 = psParent->titleArea->y() + psParent->titleArea->height();
+		psWidget->setGeometry(0, y0, psParent->width(), 30);
+	});
+
+	auto grid = std::make_shared<GridLayout>();
+	grid_allocation::slot row(0);
+
+	// Add Difficulty selector widget
+	std::vector<DIFFICULTY_LEVEL> availableDifficultyLevels = { DL_SUPER_EASY, DL_EASY, DL_NORMAL, DL_HARD, DL_INSANE };
+	if (modInfo.has_value() && !modInfo.value().difficultyLevels.empty())
+	{
+		std::vector<DIFFICULTY_LEVEL> filtered;
+		std::copy_if(availableDifficultyLevels.begin(), availableDifficultyLevels.end(), std::back_inserter(filtered), [&](DIFFICULTY_LEVEL lvl){ return modInfo.value().difficultyLevels.count(lvl) > 0; } );
```

Comment:
```suggestion
		std::copy_if(availableDifficultyLevels.begin(), availableDifficultyLevels.end(), std::back_inserter(filtered), [&modInfo](DIFFICULTY_LEVEL lvl){ return modInfo.value().difficultyLevels.count(lvl) > 0; } );
```
otherwise you'll add a lot of unneeded captures to the predicate.